src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -350,20 +341,14 @@ void goto_convertt::remove_function_call(
 
   // get name of function, if available
 
-  if(expr.id()!=ID_side_effect ||
-     expr.get(ID_statement)!=ID_function_call)
-  {
-    error().source_location=expr.find_source_location();
-    error() << "expected function call" << eom;
-    throw 0;
-  }
+  DATA_INVARIANT(
+    expr.id() == ID_side_effect && expr.get(ID_statement) == ID_function_call,
+    expr.find_source_location().as_string() + ": expected function call");
 
-  if(expr.operands().empty())
-  {
-    error().source_location=expr.find_source_location();
-    error() << "function_call expects at least one operand" << eom;
-    throw 0;
-  }
+  DATA_INVARIANT(
+    !expr.operands().empty(),
+    expr.find_source_location().as_string() +
+      ": function_call expects at least one operand");
```

Comment:
Above we require exactly two operands - could both of these `DATA_INVARIANT`s be fused into one, at the beginning of the method?

Reply:
Yes, I opted to go with the stricter version (the one requiring exactly two operands) of this invariant, given that I find that both operands are being used at various points of this function.